symengine/sets.h

```diff
@@ -0,0 +1,62 @@
+/**
+ *  \file sets.h
+ *  Polynomial Manipulation
+ *
+ **/
+#ifndef SYMENGINE_SETS_H
+#define SYMENGINE_SETS_H
+
+#include <symengine/basic.h>
+#include <symengine/functions.h>
+#include <symengine/number.h>
+#include <symengine/rational.h>
+#include <symengine/complex.h>
+
+
+namespace SymEngine {
+
+class Interval : public Basic  {
```

Comment:
`pow` makes perfect sense.
Think of Interval as a number x s.t. `a < x < b`. Then pow(i, n) would give the range of `x**n`. 

Addition of `a < x < b` and `c < y < d` would give the range of `x+y` which is `a+c, b+d`